### PR TITLE
fix(chart table in dashboard): improve screen reading of table

### PR DESCRIPTION
**Pull request by nicocarlier**
### SUMMARY
Enhancing Accessibility with Unified Table Structure and Sticky Header

This PR addresses accessibility concerns by merging the header and body of our tables into a single unified structure. This change significantly improves screen reader compatibility, ensuring a more inclusive user experience. Previously, the header and body of a chart were seperate tables. Visually this worked but caused issues with screen reader functionality.

**Key Change:**
- The table now consists of a single, unified structure, as opposed to separate elements for the header and body. This enhances accessibility, particularly for screen reader users.
- The header is made sticky using CSS, allowing it to remain visible while the table body is scrollable.

**Technical Challenge:**
A notable challenge was maintaining a scrollable body while keeping the header fixed. Setting overflow directly on **`tbody`** proved unfeasible. Wrapping **`tbody`** in a div with an overflow disrupted the column grouping, as the table would then interpret the entire **`tbody`** as a single column. The implemented solution achieves the desired layout without compromising the table's integrity or accessibility.

### TESTING INSTRUCTIONS

1. Navigate to a page featuring the modified table structure.
2. Use a screen reader to ensure that the table content is read correctly and in a logical order and doesn't read end of table early.
3. Scroll through the table to check that the header remains fixed while the body is scrollable.
4. Visually inspect the table across different browsers for consistent rendering.

### ADDITIONAL INFORMATION

- [x] Has associated issue: https://github.com/apache/superset/issues/26177
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by eschutho** [comment]
@ncar285 looks like there are some linting errors that are causing the tests to fail. I can comment on some in the PR, but you can also run `npm run lint-fix` from superset-frontend folder and that will catch some of the indentation ones.

File: superset-frontend/plugins/plugin-chart-table/src/DataTable/hooks/useSticky.tsx [hunk 1]
```diff
@@ -315,11 +318,12 @@ function StickyWrap({
...
         width: maxWidth,
         height: sticky.realHeight || maxHeight,
-        overflow: 'hidden',
+        overflow: 'auto',
+        padding: '0',
       }}
     >
-      {headerTable}
-      {bodyTable}
+      {/* {headerTable} */}
```
Comment (eschutho) [inline superset-frontend/plugins/plugin-chart-table/src/DataTable/hooks/useSticky.tsx:325]:
```suggestion
```

File: superset-frontend/plugins/plugin-chart-table/src/Styles.tsx [hunk 2]
```diff
@@ -32,10 +32,12 @@ export default styled.div`
...
       min-width: 4.3em;
     }
-
     thead > tr > th {
+      position: sticky;
+      top: -1px;
       padding-right: 0;
-      position: relative;
+      z-index: 100;
+      border-bottom: 2px solid #e0e0e0;
```
Comment (eschutho) [inline superset-frontend/plugins/plugin-chart-table/src/Styles.tsx:40]:
```suggestion
      border-bottom: ${theme.gridUnit /2}px solid ${theme.colors.grayscale.light2};
```
you'll want to use the theme here for styling when possible

File: superset-frontend/plugins/plugin-chart-table/src/DataTable/hooks/useSticky.tsx [hunk 3]
```diff
@@ -215,7 +217,9 @@ function StickyWrap({
   let sizerTable: ReactElement | undefined;
   let headerTable: ReactElement | undefined;
   let footerTable: ReactElement | undefined;
-  let bodyTable: ReactElement | undefined;
+  // let bodyTable: ReactElement | undefined;
```
Comment (eschutho) [inline superset-frontend/plugins/plugin-chart-table/src/DataTable/hooks/useSticky.tsx:220]:
```suggestion
  // let bodyTable: ReactElement | undefined;
```

File: superset-frontend/plugins/plugin-chart-table/src/DataTable/hooks/useSticky.tsx [hunk 4]
```diff
@@ -16,6 +16,7 @@
  * specific language governing permissions and limitations
  * under the License.
  */
+
```
Comment (eschutho) [inline superset-frontend/plugins/plugin-chart-table/src/DataTable/hooks/useSticky.tsx:19]:
```suggestion

```

File: superset-frontend/plugins/plugin-chart-table/src/DataTable/DataTable.tsx [hunk 5]
```diff
@@ -16,6 +16,7 @@
  * specific language governing permissions and limitations
  * under the License.
  */
+
```
Comment (eschutho) [inline superset-frontend/plugins/plugin-chart-table/src/DataTable/DataTable.tsx:19]:
```suggestion

```

**Review by eschutho — CHANGES_REQUESTED** [review]
Thanks @ncar285! Looks really good. Just some small suggestions for linting and cleanup.

**Review by eschutho — APPROVED** [review]
This looks great @ncar285. I'll leave the PR open for testing, and then we'll merge. Thank you!

**Comment by kgabryje** [comment]
@ncar285 @eschutho I think after this PR the Totals are not sticky anymore 🙁 

https://github.com/apache/superset/assets/15073128/e58fdb5e-7702-4cce-a318-2a876a091a49

**Pull request merged**